source/dub/compilers/ldc.d

```diff
@@ -234,7 +231,9 @@ config    /etc/ldc2.conf (x86_64-pc-linux-gnu)
 			case TargetType.executable: break;
 			case TargetType.library:
 			case TargetType.staticLibrary:
-				settings.addDFlags("-lib");
+				// -oq: name object files uniquely (so the files don't collide)
+				// -cleanup-obj: remove object files after archiving to static lib, and put them in a unique temp directory
+				settings.addDFlags("-lib", "-oq", "-cleanup-obj");
```

Comment:
are the new flags not required for the dynamicLibrary case below?

Reply:
Nope; the object files for executables and shared libs are persisted/kept around, by dub and the compilers alike. Static libs with DMD are a special case, as it writes the static lib directly itself, without creating separate object files and then archiving them, like LDC and GDC. `ldmd2 -lib` also injects these 2 extra flags for the `ldc2` cmdline.